### PR TITLE
add spark sdk/cli sample

### PR DESCRIPTION
**Pull request by YingChen1996**
# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

File: sdk/jobs/pipelines/1j_pipeline_with_spark_nodes/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1j_pipeline_with_spark_nodes/README.md:1]: Let's use 1i_xxx

File: cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      file: entry.py # file path of the entry file relative to the code root folder
+
+    py_files:
+      - utils.zip
+    jars:
+      - scalaproj.jar
+
+    files:
+      - my_files.txt
+    #archives:
```
Comment (wangchao1230) [inline cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml:36]: remove archives if not effectively used

File: cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    path: ./dataset/iris.csv
+    mode: direct
+
+outputs:
+  output:
+    type: uri_folder
+    mode: direct
+
+jobs:
+  spark_job:
```
Comment (wangchao1230) [inline cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml:17]: I think we should have a sample with node that at least train a model

File: sdk/jobs/pipelines/1j_pipeline_with_spark_nodes/pipeline_with_spark_nodes.ipynb [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+   "metadata": {},
+   "source": [
+    "# Next Steps\n",
+    "You can see further examples of running a pipeline job [here](../)"
+   ]
+  }
+ ],
+ "metadata": {
+  "description": {
+   "description": "Create pipeline with parallel node to do batch inference"
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1j_pipeline_with_spark_nodes/pipeline_with_spark_nodes.ipynb:225]: update this description

File: cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+type: pipeline
+
+description: 'submit a spark job using component sdk'
```
Comment (wangchao1230) [inline cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml:3]: No `compoennt sdk`, update: 
submit a pipeline with spark job

File: cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml (hunk 2 above)
Comment (YingChen1996) [reply cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml:36]: removed

File: cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml (hunk 5 above)
Comment (YingChen1996) [reply cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml:3]: updated

File: sdk/jobs/pipelines/1j_pipeline_with_spark_nodes/pipeline_with_spark_nodes.ipynb (hunk 4 above)
Comment (YingChen1996) [reply sdk/jobs/pipelines/1j_pipeline_with_spark_nodes/pipeline_with_spark_nodes.ipynb]: updated

File: cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml (hunk 3 above)
Comment (YingChen1996) [reply cli/jobs/pipelines/iris-add-column-and-word-count-using-spark/pipeline.yml:17]: add spark kmeans node

File: sdk/jobs/pipelines/1j_pipeline_with_spark_nodes/pipeline_with_spark_nodes.ipynb [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+   "metadata": {
+    "name": "build-pipeline"
+   },
+   "outputs": [],
+   "source": [
+    "synapse_compute_name = \"spark31\"\n",
+    "\n",
+    "@pipeline(description=\"submit a spark job using component sdk\")\n",
+    "def spark_pipeline_from_yaml(iris_data):\n",
+    "    add_new_column_and_word_count_node = add_new_column_and_word_count(file_input1=iris_data, file_input2=iris_data)\n",
```
[the remark this answers is not in the document]
Comment (YingChen1996) [reply sdk/jobs/pipelines/1j_pipeline_with_spark_nodes/pipeline_with_spark_nodes.ipynb]: add two node pipeline, one spark node is to train model and command node is to print result

File: sdk/jobs/pipelines/1i_pipeline_with_spark_nodes/components/component.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+$schema: http://azureml/sdk-2-0/SparkComponent.json
+name: kmeans_spark_component
+type: spark
+version: 1
+display_name: Aml Spark dataset test module
```
Comment (wangchao1230) [inline sdk/jobs/pipelines/1i_pipeline_with_spark_nodes/components/component.yml:5]: update the display_name and description

File: cli/jobs/pipelines/shakespear-sample-and-word-count-using-spark/component/sample_word.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
+$schema: http://azureml/sdk-2-0/SparkComponent.json
+name: sample_spark_component
+type: spark
+version: 1
+display_name: Aml Spark dataset sample test module
```
Comment (wangchao1230) [inline cli/jobs/pipelines/shakespear-sample-and-word-count-using-spark/component/sample_word.yml:5]: refine the display_name and description

**Pull request closed without merging**